### PR TITLE
Converge melos usage and CI scripts & add code coverage

### PR DESCRIPTION
**Pull request by kuhnroyal**
The idea of this PR is to make it easier to run tests locally based on melos while also adding code coverage reporting.

* move all scripts to melos and make them available for usage locally
* automatically generation coverage information for all Dart tests (vm, chrome, firefox)
* print coverage results to CLI for stable SDK
* only run analyze/format/publish on Stable SDK, this way we can skip the patching for format/analyze for older SDK versions which actually makes no sense since it is not being released

I want to further extend the coverage reporting by using https://github.com/clearlyip/code-coverage-report-action.
This action would run for each push/merge to the main branch an generate a constant coverage baseline for the main branch. The baseline is uploaded as workflow artifact and as such available to pull request which can then display the change in coverage.

### New Pull Request Checklist

- [ ] I have read the [Documentation](https://pub.dev/documentation/dio/latest/)
- [ ] I have searched for a similar pull request in the [project](https://github.com/cfug/dio/pulls) and found none
- [ ] I have updated this branch with the latest `main` branch to avoid conflicts (via merge from master or rebase)
- [ ] I have added the required tests to prove the fix/feature I'm adding
- [ ] I have updated the documentation (if necessary)
- [ ] I have run the tests without failures
- [ ] I have updated the `CHANGELOG.md` in the corresponding package

### CODE REVIEW

**Comment by kuhnroyal** [comment]
There are currently random test timeouts due to network/httpbun but I think this is good to go.
There is already a job summary for the stable matrix job which displays the coverage.
More information for coverage diffs will hopefully be available once the workflow on the main branch runs.

**Comment by kuhnroyal** [comment]
@cfug-dev Looking for feedback on this change. I have a follow-up which moves most of the tests to a local httpun docker container in CI, which is based on this.

File: .github/workflows/coverage_base.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        run: melos run test:web
+        env:
+          TEST_PLATFORM: firefox
+      - name: '[Verify step] Test Flutter packages'
+        run: melos run test:flutter
+      - name: '[Coverage] Generate report'
+        run: melos run coverage:combine
+      - uses: clearlyip/code-coverage-report-action@v4
+        with:
+          filename: 'coverage/cobertura.xml'
```
Comment (AlexV525) [inline .github/workflows/coverage_base.yml:52]: Looks like a proper EOF is needed.

File: .github/workflows/coverage_base.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  workflow_dispatch:
+  # Runs at 00:00, on day 1 of the month (every ~30 days)
+  schedule:
+    - cron: '0 0 1 * *'
+  push:
+    branches:
+      - main
+
+jobs:
+  cron:
```
Comment (AlexV525) [inline .github/workflows/coverage_base.yml:21]: Maybe rename to generate?

File: scripts/melos_packages.dart [hunk 3]
```diff
@@ -6,6 +6,10 @@ import 'package:melos/melos.dart'
 import 'package:pub_semver/pub_semver.dart';
 import 'package:yaml/yaml.dart';
 
+/// Writes a `.melos_packages` file to the root of the workspace with the
+/// packages that are compatible with the current Dart SDK version.
+/// This is useful for CI scripts that need to know which packages to run
+/// melos for ny using the `MELOS_PACKAGES` environment variable.
```
Comment (AlexV525) [inline scripts/melos_packages.dart:12]: `ny` a typo?

**Comment by AlexV525** [comment]
This look fantastic and those scripts become elegant! Two more things might be added to the README too:
1. How do contributors run tests and what matrixs should they verify locally?
2. The current coverage of each packages.

**Comment by kuhnroyal** [comment]
Added individual melos scripts for firefox/chrome and renamed `test:dart` to `test:vm`.

**Comment by kuhnroyal** [comment]
I added a `CONTRIBUTING` file with some basic guidelines and the help of Copilot.

**Comment by kuhnroyal** [comment]
I don't want to add anything about coverage yet, maybe when it has settled and is working as expected in PRs.

**Comment by AlexV525** [comment]
> I added a `CONTRIBUTING` file with some basic guidelines and the help of Copilot.

Could we also have a Chinese version? The library is widely used in the Chinese community. Adding the translation within the same file or make a separate file both look good to me.

**Review by AlexV525 — APPROVED** [review]

**Comment by kuhnroyal** [comment]
Would you mind adding a `CONTRIBUTING-ZH.md` file?

**Comment by AlexV525** [comment]
> Would you mind adding a `CONTRIBUTING-ZH.md` file?

That would be nice too.

**Comment by AlexV525** [comment]
This should also address #2136.

**Comment by kuhnroyal** [comment]
Can you write/translate that file or should I just ask ChatGPT? :D

**Comment by AlexV525** [comment]
> Can you write/translate that file or should I just ask ChatGPT? :D

Yeah please go ahead and let GPT helps

**Comment by kuhnroyal** [comment]
> This should also address #2136.

That is this PR :)

Add a Chinese version, can you check?

**Review by AlexV525 — APPROVED** [review]
👍🏻

**Comment by AlexV525** [comment]
> > This should also address #2136.
> 
> That is this PR :)

Ah, pasted the wrong number. It should be #1988

**Pull request merged**